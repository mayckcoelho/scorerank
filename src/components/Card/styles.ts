import styled from 'styled-components/native';

export const CardStyled = styled.View`
    margin: 20px 0;
    padding: 20px;
    box-shadow: 0 5px 8px rgba(0, 0, 0, .3);
    border-radius: 6px;

    width: 95%;
`