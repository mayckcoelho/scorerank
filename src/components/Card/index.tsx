import React from 'react';

import { CardStyled } from './styles';

interface Props {
    background?: string;
}

const Card: React.FC<Props> = ({ background, children }) => (
    <CardStyled style={{ backgroundColor: background ?? '#fff' }}>
        {children}
    </CardStyled>
)

export default Card;