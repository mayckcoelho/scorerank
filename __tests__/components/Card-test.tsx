import React from 'react';
import Card from '../../src/components/Card';
import renderer from 'react-test-renderer';

describe('Card should render correctly', () => {
    it('match snapshot', () => {
        const card = renderer
            .create(<Card background="#fff" />)
            .toJSON();
        expect(card).toMatchSnapshot();
    });
})