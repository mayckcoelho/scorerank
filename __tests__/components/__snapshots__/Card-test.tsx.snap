// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card should render correctly match snapshot 1`] = `
<View
  style={
    Array [
      Object {
        "borderRadius": 6,
        "marginBottom": 20,
        "marginLeft": 0,
        "marginRight": 0,
        "marginTop": 20,
        "paddingBottom": 20,
        "paddingLeft": 20,
        "paddingRight": 20,
        "paddingTop": 20,
        "shadowColor": "rgba(0, 0, 0, .3)",
        "shadowOffset": Object {
          "height": 5,
          "width": 0,
        },
        "shadowOpacity": 1,
        "shadowRadius": 8,
        "width": "95%",
      },
      Object {
        "backgroundColor": "#fff",
      },
    ]
  }
/>
`;
